bash_functions.sh

```diff
@@ -193,6 +194,32 @@ setup_dnsmasq_hostnames() {
     fi
 }
 
+setup_cache_size() {
+    local warning="WARNING: Custom CACHE_SIZE not used"
+    local dnsmasq_pihole_01_location="/etc/dnsmasq.d/01-pihole.conf"
+    # Quietly exit early for empty or default
+    if [[ -z "${1}" || "${1}" == '10000' ]] ; then return ; fi
+
+    if [[ "${DNSSEC}" == "true" ]] ; then
+        echo "$warning - Cannot change cache size if DNSSEC is enabled"
+        return
+    fi
+
+    if ! echo $1 | grep -q '^[0-9][0-9]*$' ; then
```

Comment:
Nitpick: Is there a reason for the regex like this or would `^[0-9]+$` work? Not marking as a change request, but if there's no objection to simplifying it then would be nice to have.

Reply:
That's a nice find. I copied this line from https://github.com/pi-hole/docker-pi-hole/blob/f4814935b9c0c531c415b3c9ae58af97466f30cd/bash_functions.sh#L258, but looking closely at it it definitely would be better to simplify it. Changed.